### ISSUE TITLE
ColumnToggler: does not work with keyboard

### ISSUE THREAD

**Issue by klippelU** [primefaces/primefaces#13042]
### Describe the bug

At PF14 there is a new feature at column toggler to select all/no column of table. This works fine with mouse selection but - also provable in showcase - there will be a fault, if you try to select the 'all' checkbox by using space bar.

![Column toggler](https://github.com/user-attachments/assets/58b1e308-68e3-4d98-b9e0-cbc30affbcb1)

### PrimeFaces edition

None

### PrimeFaces version

14.0.6

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

11

**Comment by melloware** [comment]
Definitely a bug!

**Comment by klippelU** [comment]
I test this today with PF14.0.9. In my case it still not works as expected. 

Now is possible for me, if the select-all-box is marked (by mouse) to toggle all by space bar. But I'm not able to select the box by using the Tab key. In my case it jumps upward in 2-time-steps meaning it skips one checkbox after another so the select-all-checkbox is one of them, who were skiped. 

Move Tab key in reverse direction (shift-tab) is not possible anymore. The key command seems to be blocked.

Could you confirm this?

**Comment by melloware** [comment]
Confirmed TABBING is not right.  Reopening.

**Comment by melloware** [comment]
you can see it in action here: https://quarkus-faces-melloware-8a6a34c1.koyeb.app/ui/data/datatable/columnToggler.xhtml  

Give it a try

**Comment by klippelU** [comment]
This is helpful. Thanks.

First things first: the function of Column Toggler works as expected. Very good.

There are only two things, that I would mention:
- At your ColumnToggler box there is a column with name 'Code', but I miss the heading 'Code' in DataTable, if selected.
- Second is about accessibility. I checked the Toggler with screen reader. It reads the 'all' switch as 'All items unselected'. This might be missleading. Because some of the columns are selected. It should be named somehow as 'Unselect all items', or maybe 'Switch Unselect all items', to explain the correct functionality. Of course it maybe possible to override the aria-labels, but would be nice, if it works out of the box.

**Comment by melloware** [comment]
its actually backwards too.  When the checkbox is UNCHECKED it should be "Select All Items" as that is what the action does.  When its checked it should be "Unselect All Items".
